cmd/catalog-importer/cmd/sync.go

```diff
@@ -310,12 +317,55 @@ createCatalogType:
 				return errors.Wrap(err, fmt.Sprintf("outputs.%d (type_name='%s')", idx, outputType.TypeName))
 			}
 
-			logger.Log("msg", "reconciling catalog entries", "output", outputType.TypeName)
-			catalogType := catalogTypesByOutput[outputType.TypeName]
+			// This can be reused for both model and enum types.
+			entriesClient := newEntriesClient(cl, existingCatalogTypes, opt.DryRun)
 
-			err = reconcile.Entries(ctx, logger, newEntriesClient(cl, existingCatalogTypes, opt.DryRun), catalogType, entryModels, newEntriesProgress(!opt.DryRun))
-			if err != nil {
-				return errors.Wrap(err, fmt.Sprintf("outputs (type_name = '%s'): reconciling catalog entries", outputType.TypeName))
+			{
+				logger.Log("msg", "reconciling catalog entries", "output", outputType.TypeName)
+				catalogType := catalogTypesByOutput[outputType.TypeName]
+
+				err = reconcile.Entries(ctx, logger, entriesClient, catalogType, entryModels, newEntriesProgress(!opt.DryRun))
+				if err != nil {
+					return errors.Wrap(err, fmt.Sprintf("outputs (type_name = '%s'): reconciling catalog entries", outputType.TypeName))
+				}
+			}
+
+			// Process enum attributes, which require generating from the result of the parent
+			// model's attribute.
+			_, enumModels := output.MarshalType(outputType)
+			for _, enumModel := range enumModels {
+				// We've got an enum attribute, which means we need to sync the enum values.
+				valueSet := map[string]bool{}
+				for _, entry := range entryModels {
+					value := entry.AttributeValues[enumModel.EnumAttributeID.String]
+					if value.Value != nil {
+						valueSet[*value.Value.Literal] = true
+					}
+					if value.ArrayValue != nil {
+						for _, elementValue := range *value.ArrayValue {
+							valueSet[*elementValue.Literal] = true
+						}
+					}
+				}
+
+				enumModels := []*output.CatalogEntryModel{}
+				for value := range valueSet {
+					enumModels = append(enumModels, &output.CatalogEntryModel{
+						ExternalID:      value,
+						Name:            value,
+						Aliases:         []string{},
```

Comment:
Should we add the `value` as an alias or external ID here? I thinking if I have an enum with `language-ruby` then I might want to go and give that a pretty name of "Ruby" in the future, while keeping that as the identifier.

Reply:
I think the idea would be that the "name" (human name) starts out life as the inhumane name, and then if you edit the value in the catalog (e.g. via the UI) then it doesn't get wipped out? Not sure if that's possible though.